spec/text-editor-presenter-spec.coffee

```diff
@@ -2227,6 +2099,407 @@ describe "TextEditorPresenter", ->
+            expect(newGutterDecorationState[decoration2.id]).toBeUndefined()
+            expect(newGutterDecorationState[decoration3.id]).toBeUndefined()
+            oldGutterDecorationState = decorationStateForGutterName(presenter, 'test-gutter')
+            expect(oldGutterDecorationState[decoration1.id]).toBeUndefined()
+            expect(oldGutterDecorationState[decoration2.id].top).toBeDefined()
+            expect(oldGutterDecorationState[decoration3.id]).toBeUndefined()
+
+        it "updates when the editor's mini state changes, and is cleared when the editor is mini", ->
+          expectStateUpdate presenter, -> editor.setMini(true)
+          decorationState = decorationStateForGutterName(presenter, 'test-gutter')
+          expect(decorationState).toBeUndefined()
+
+          # The decorations should return to the original state.
+          expectStateUpdate presenter, -> editor.setMini(false)
+          decorationState = decorationStateForGutterName(presenter, 'test-gutter')
+          expect(decorationState[decoration1.id].top).toBeDefined()
+          expect(decorationState[decoration2.id].top).toBeDefined()
+          expect(decorationState[decoration3.id]).toBeUndefined()
+
+        it "updates when a gutter's visibility changes, and is cleared when the gutter is not visible", ->
+          expectStateUpdate presenter, -> gutter.hide()
+          decorationState = decorationStateForGutterName(presenter, 'test-gutter')
+          expect(decorationState[decoration1.id]).toBeUndefined()
+          expect(decorationState[decoration2.id]).toBeUndefined()
+          expect(decorationState[decoration3.id]).toBeUndefined()
+
+          # The decorations should return to the original state.
+          expectStateUpdate presenter, -> gutter.show()
+          decorationState = decorationStateForGutterName(presenter, 'test-gutter')
+          expect(decorationState[decoration1.id].top).toBeDefined()
+          expect(decorationState[decoration2.id].top).toBeDefined()
+          expect(decorationState[decoration3.id]).toBeUndefined()
+
+        it "updates when a gutter is added to the editor", ->
+          decorationParams =
+            type: 'gutter'
+            gutterName: 'test-gutter-2'
+            class: 'test-class'
+          marker4 = editor.markBufferRange([[0,0],[1,0]])
+          decoration4 = editor.decorateMarker(marker4, decorationParams)
+          expectStateUpdate presenter, -> editor.addGutter({name: 'test-gutter-2'})
+
+          decorationState = decorationStateForGutterName(presenter, 'test-gutter-2')
+          expect(decorationState[decoration1.id]).toBeUndefined()
+          expect(decorationState[decoration2.id]).toBeUndefined()
+          expect(decorationState[decoration3.id]).toBeUndefined()
+          expect(decorationState[decoration4.id].top).toBeDefined()
+
+        it "does not update when a gutter is destroyed (because it's unnecessary)", ->
```

Comment:
If you'd rather remove them (and sounds like it :p), I can do it. It's just unnecessary because the next time any decoration state changes (or if the gutter is re-added), that will trigger an update that clears these old decorations.